File: move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        '''
        Given a list nums of ints, moves all the zeroes to the end
        Performs the move in-place
        '''
        count = 0
        for i in range(len(nums)):
            if nums[i - count] == 0:
                nums.pop(i - count)
                count += 1
        while count > 0:
            nums.append(0)
            count -= 1
        return None

File: move-zeroes/move-zeroes.java
class Solution 
{
    public void moveZeroes(int[] nums) 
    {
        // Move all digits to front of the array
        int lastZero = 0;
        for (int i = 0; i < nums.length; i++)
            if (nums[i] != 0)
                nums[lastZero++] = nums[i];
        
        // Insert zeroes once last digit found
        for (int i = lastZero; i < nums.length; i++)
            nums[i] = 0;
    }
}